modules/reactor/include/meshgenerators/PeripheralRingMeshGenerator.h

```diff
@@ -0,0 +1,85 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+#include "PolygonMeshGeneratorBase.h"
+#include "MooseEnum.h"
+#include "MooseMeshUtils.h"
+#include "MeshMetaDataInterface.h"
+
+/**
+ * This PeripheralRingMeshGenerator object adds a circular peripheral region to the input mesh.
+ */
+class PeripheralRingMeshGenerator : public PolygonMeshGeneratorBase
+{
+public:
+  static InputParameters validParams();
+
+  PeripheralRingMeshGenerator(const InputParameters & parameters);
+
+  std::unique_ptr<MeshBase> generate() override;
+
+protected:
+  /// Name of the mesh generator to get the input mesh
+  const MeshGeneratorName _input_name;
+  /// Number of layers of elements of the peripheral region in radial direction
+  const unsigned int _peripheral_layer_num;
+  /// Radius of the peripheral region's outer circular boundary
+  const Real _peripheral_ring_radius;
+  /// Volume preserving function is optional
+  const bool _preserve_volumes;
+  /// Name of the external boundary of the input mesh
+  const BoundaryName _input_mesh_external_boundary;
+  /// Subdomain ID of the added peripheral region
+  const subdomain_id_type _peripheral_ring_block_id;
+  /// Subdomain name of the added peripheral region
+  const SubdomainName _peripheral_ring_block_name;
+  /// ID of the new external boundary
+  const boundary_id_type _external_boundary_id;
+  /// Name of the new external boundary
+  const BoundaryName _external_boundary_name;
+  /// Reference to input mesh pointer
+  std::unique_ptr<MeshBase> & _mesh;

```

Comment:
```suggestion
  std::unique_ptr<MeshBase> & _input;
```
it's the name used everywhere else. 
If you could rename it in AzimuthalBlockSplitGenerator.h would be good too

Reply:
will do